### PR TITLE
FOGL-2980: Automate testing of Modbus RTU

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
test execution:
/FogLAMP/tests/system/python/e2e$ pytest -s -vv test_e2e_modbus_c_rtu_pi.py --modbus-serial-port=<VALUE> --modbus-baudrate <other_pi_config>

Result:
===== 1 passed in 35.85 seconds ===

### CODE REVIEW

File: tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+__version__ = "${VERSION}"
+
+SOUTH_PLUGIN = "modbus-c"
+PLUGIN_NAME = "ModbusC"
+SVC_NAME = "modbus-c"
+ASSET_NAME_1 = "adam4015"
+ASSET_NAME_2 = "adam4017"
+
+
+class TestE2EModbusCMTUPI:
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py]: TestE2EModbusC_RTU_PI

File: tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                          modbus_serial_port, modbus_baudrate):
+        """ This fixture clone a south and north repo and starts both south and north instance
+
+            reset_and_start_foglamp: Fixture that resets and starts foglamp, no explicit invocation, called at start
+            add_south: Fixture that adds a south service with given configuration with enabled or disabled mode
+            remove_directories: Fixture that remove directories created during the tests
+        """
+        self.check_connect(modbus_serial_port, modbus_baudrate)
+        cfg = {"protocol": {"value": "RTU"}, "asset": {"value": "modbus"}, "address": {"value": "127.0.0.1"},
+               "port": {"value": "{}".format("2222")}, "device": {"value": modbus_serial_port},
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py]: address and port should NOT be needed. 

Use the given `--modbus-baudrate` or let it be default (and remove from conftest / arg )

File: tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+""" Test system/python/e2e/test_e2e_modbus_c_rtu_pi.py
+
+"""
+
+import http.client
+import json
+import time
+import pytest
+import serial
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py]: no module serial.

looks we need to add pyserial as test dep?

Comment (Singhal-Vaibhav) [reply tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py]: added

File: tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+SOUTH_PLUGIN = "modbus-c"
+PLUGIN_NAME = "ModbusC"
+SVC_NAME = "modbus-c"
+ASSET_NAME_1 = "adam4015"
+ASSET_NAME_2 = "adam4017"
+
+
+class TestE2EModbusC_RTU_PI:
+    def check_connect(self, modbus_serial_port, modbus_baudrate):
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_modbus_c_rtu_pi.py]: modbus_baudrate has no value here. BTW

**Pull request merged**